frontend/src/pages/SelectRoom.jsx

```diff
@@ -1,62 +1,57 @@
 import { useLocation } from 'react-router-dom';
 import { useNavigate } from 'react-router-dom';
 import RoomCard from "../components/RoomCard";
+import {getAvailableRooms} from "../services/roomsService.js";
+import {useEffect, useState} from "react";
 
-export default function SelectRoom()
-{
+export default function SelectRoom() {
+  const [rooms, setRooms] = useState([]);
+  const [loading, setLoading] = useState(true);
   const navigate = useNavigate();
   const location = useLocation();
   const { checkInDate, checkOutDate, numPeople } = location.state;
 
-  const rooms = [{
-    Beds: [
-      {
-        Size: "Queen",
-        Capacity: "2"
+  useEffect(() => {
+    async function fetchRooms() {
+      try {
+        const availableRooms = await getAvailableRooms(checkInDate, checkOutDate);
```

Comment:
If there are no rooms available on the requested dates, a message should be displayed warning of this and to try other dates, it should be in the middle as well as the loading.

Reply:
Done :D